src/modeler/AutoRest.Swagger/Validation/NetworkInterfacesApiWarning.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+using System.Text.RegularExpressions;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class NetworkInterfacesApiWarning : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
+    {
+        private const string MicrosoftComputeApi = "Microsoft.Compute";
+        private const string MicrosoftNetworkApi = "Microsoft.Network";
+        /// <summary>
+        /// This rule passes if the paths contain reference to either Microsoft.Network Apis or 
+        /// Microsoft.Compute Apis but not both
+        /// </summary>
+        /// <param name="paths"></param>
+        /// <returns></returns>
+        public override bool IsValid(Dictionary<string, Dictionary<string, Operation>> paths)
+        {
+            Regex ApiRegExp = new Regex(@"(/.+)+/((M|m)icrosoft.(C|c)ompute|(M|m)icrosoft.(N|n)etwork)/.+");
```

Comment:
`MICROSOFT.COMPUTE` will pass thru that. 

Perhaps we want a case insensitive regex?